nupic/research/frameworks/dendrites/modules/dendritic_layers.py

```diff
@@ -173,43 +194,24 @@ class GatingDendriticLayer2d(DendriticLayer2dBase):
     multiplied by a single value computed via dendrites.
     """
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.dendritic_gate_2d = DendriticGate2d()
```

Comment:
Seems this class is a lot similar to GatingDendriticLayer1d, whith the exception of the type of Gate it receives, whether 1d or 2d. Should that really be 2 separate classes?

Reply:
If I'm not mistaken, we want to follow the convention of having a separate class for 1D and 2D operations, as followed by `SparseWeights` vs `SparseWeights2D`, and also `torch.nn.Conv1d` vs `torch.nn.Conv2d`.

Reply:
There's one `DendriticLayer`class per `apply_dendrites` function essentially. That is the defining feature of these classes.

Reply:
Note `GatingDendriticLayer1d` and `GatingDendriticLayer2d` inherit from `DendriticLayerBase1d` and `DendriticLayerBase2d` respectively, which also differ in whether they are subclasses from `SparseWeights` or `SparseWeights2d`.